config/routes.rb

```diff
@@ -5,8 +5,8 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
-  get '/', to: 'users#user_data'
-  get '/users/:id', to: 'users#show', as: 'user'
-  get 'users/:user_id/posts', to: 'users#show_posts', as: 'user_posts'
-  get 'users/:user_id/posts/:post_id', to: 'users#show_post', as: 'user_post'
+  get '/users', to: 'users#index'
+  get '/users/:id', to: 'users#show'
+  get 'users/:user_id/posts', to: 'posts#index'
+  get 'users/:user_id/posts/:post_id', to: 'posts#show'
```

Comment:
## Optional
- [ ] I suggest that you use `Resource` routing because it allows you to quickly declare all of the common routes for a given resourceful controller.
```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```